examples/chip-tool/commands/payload/SetupPayloadVerhoeff.cpp

```diff
@@ -0,0 +1,78 @@
+/*
+ *   Copyright (c) 2020 Project CHIP Authors
+ *   All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+
+#include "SetupPayloadVerhoeff.h"
+#include "SetupPayloadParseCommand.h"
+#include <setup_payload/SetupPayload.h>
+#include <support/verhoeff/Verhoeff.h>
+
+using namespace ::chip;
+
+CHIP_ERROR SetupPayloadVerhoeffVerify::Run()
+{
+    std::string codeString(mSetupCode);
+    bool result;
+
+    bool isQRCode = SetupPayloadParseCommand::IsQRCode(codeString);
+
+    VerifyOrReturnError(!isQRCode, CHIP_ERROR_NOT_IMPLEMENTED);
+    VerifyOrReturnError(codeString.length() > mPos && mPos >= 0, CHIP_ERROR_INVALID_STRING_LENGTH);
```

Comment:
could we do the codeString length check before calling IsQRCode? Ordering wise it looks like this is the simplest and should be the first thing to validate.

Reply:
Thank you for the review. Updated.